build-and-test.ps1

```diff
@@ -0,0 +1,46 @@
+[cmdletbinding()]
+param(
+    [string]$VersionFilter,
+    [string]$OSFilter,
+    [switch]$IsDryRun
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+$(docker version) | ForEach-Object { Write-Host "$_" }
```

Comment:
Docker version is emitted here and in ImageBuilder. Either move this statement into the `if` block where ImageBuilder image is pulled (or) remove version emit in ImageBuilder itself.  Any preferences?

Reply:
Agree. My thinking earlier was - Docker is a prerequisite for the build-test experience.  If a prerequisite is not available, in this case Docker version command fails, then exit immediately.